scipy/stats/_continuous_distns.py

```diff
@@ -3456,8 +3456,21 @@ def _munp(self, n, a, c):
         return sc.poch(a, n*1.0/c)
 
     def _entropy(self, a, c):
-        val = sc.psi(a)
-        return a*(1-val) + 1.0/c*val + sc.gammaln(a) - np.log(abs(c))
+        def regular(a, c):
+            val = sc.psi(a)
+            A = a * (1 - val) + val / c
+            B = sc.gammaln(a) - np.log(abs(c))
+            h = A + B
+            return h
+
+        def asymptotic(a, c):
+            # using asymptotic expansions for gammaln and psi (see gh-18093)
+            return ((-np.log(a) + np.log(2) + 1 + np.log(np.pi))/2
+                    - np.log(np.abs(c)) + (a**-1)/6 - (a**-3)/90
+                    + (np.log(a) - (a**-1)/2 - (a**-2)/12 + (a**-4)/120)/c)
```

Comment:
```suggestion
            return ((-np.log(a) + np.log(2) + 1 + np.log(np.pi))/2
                    - np.log(np.abs(c)) + (a**-1.)/6 - (a**-3.)/90
                    + (np.log(a) - (a**-1.)/2 - (a**-2.)/12 + (a**-4.)/120)/c)
```
ugh this `ValueError: Integers to negative integer powers are not allowed.` is annoying.